integration/tools/validation/src/main/java/alluxio/cli/HdfsValidationTool.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli;
+
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.InstancedConfiguration;
+import alluxio.underfs.UnderFileSystemConfiguration;
+
+import com.google.common.collect.ImmutableMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A tool to validate an HDFS mount, before the path is mounted to Alluxio.
+ * */
+public class HdfsValidationTool {
```

Comment:
This is previously the `ValidateHdfsMount` class. Now it's split into two:
1. The real work logic is here.
1. The trigger is in a new `ValidateHdfsMount` class in shell/